### DISCUSSION TITLE
How to get pyright to recognize a local package-stubs directory

### DISCUSSION THREAD

**Discussion by Dr-Irv** [microsoft/pyright#3371] [General]
I need this for the pandas-stubs project we are creating, but it is more of a general question.

Let's say that I want to create a set of type stubs for package `xyz` that is installed in my environment.  The goal is to create a project `xyz-stubs` that would appear in pypi and conda that people could install.

For testing purposes, I have a my project with a `xyz-stubs` directory that has the stubs I am creating, along with a `tests` directory that has code that can test those stubs.  

From the command line, I want to run `pyright` so that it finds the local `xyz-stubs` directory to look for the stubs for `xyz`.  I can use the pyright config file option `include` to refer to `tests` so that pyright type checks the files in `tests`.  It's not clear what settings I should use to make things happen so that `pyright` will use the local `xyz-stubs` folder to check the types of the package `xyz`.

**Comment by erictraut** [comment]
Converting to a discussion topic since this is a general question rather than a bug or a feature request.

**Reply by erictraut** [reply]
Would it work to do an editable install of the stub package? That way, it will be installed in your test environment in a manner that's similar to how it will later be installed by a consumer of the stub package.

Another approach is to have your test script create a sym link in your local typings directory called `xyz` that points to the `xyz-stubs` directory.
